### PR TITLE
feat: exposing portal for webhook

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: core/api/src/app/callback/index.ts [hunk 1]
```diff
@@ -54,3 +54,11 @@ export const deleteEndpoint = async ({
   if (success instanceof Error) return success
   return success
 }
+
+export const getPortal = async (accountId: AccountId) => {
+  const callbackService = CallbackService(getCallbackServiceConfig())
+  const res = await callbackService.getPortal(accountId)
+
+  if (res instanceof Error) return res
+  return res
```
Comment (dolcalmi) [inline core/api/src/app/callback/index.ts:63]:
```suggestion
  return await callbackService.getPortal(accountId)
```

File: core/api/src/services/svix/index.ts [hunk 2]
```diff
@@ -113,7 +113,7 @@ export const CallbackService = (config: SvixConfig) => {
   }
 
   // only work for hosted svix
-  const getWebsocketPortal = async (accountId: AccountId) => {
+  const getPortal = async (accountId: AccountId) => {
```
Comment (dolcalmi) [inline core/api/src/services/svix/index.ts:116]: dont we have an interface for this? we must include return type

File: core/api/src/app/callback/index.ts [hunk 3]
```diff
@@ -54,3 +54,11 @@ export const deleteEndpoint = async ({
   if (success instanceof Error) return success
   return success
 }
+
+export const getPortal = async (accountId: AccountId) => {
```
Comment (dolcalmi) [inline core/api/src/app/callback/index.ts:58]: should not be `getPortalUrl`? also please use `{accountId}: {accountId: AccountId}` this creates `code.function.params.accountId` in span attrs

File: core/api/src/services/svix/index.ts [hunk 4]
```diff
@@ -113,7 +113,7 @@ export const CallbackService = (config: SvixConfig) => {
...
     const accountCallbackId = getAccountCallbackId(accountId)
```
Comment (dolcalmi) [inline core/api/src/services/svix/index.ts:117]: please include line 117-119 inside try block

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: core/api/src/app/callback/index.ts (hunk 3 above)
Comment (nicolasburtey) [reply core/api/src/app/callback/index.ts:58]: oh I didn't know that.

File: core/api/src/app/callback/index.ts (hunk 1 above)
Comment (nicolasburtey) [reply core/api/src/app/callback/index.ts:63]: we may not need/want the `await`?

File: core/api/src/services/svix/index.ts (hunk 4 above)
Comment (nicolasburtey) [reply core/api/src/services/svix/index.ts:117]: `getAccountCallbackId(accountId)` is just a string operation. it won't fail

File: core/api/src/services/svix/index.ts [hunk 5]
```diff
@@ -31,15 +31,36 @@ function prefixObjectKeys(
   }, {})
 }
 
-export const CallbackService = (config: SvixConfig) => {
+interface CallbackService {
```
Comment (dolcalmi) [inline core/api/src/services/svix/index.ts:34]: please move to https://github.com/GaloyMoney/galoy/blob/main/core/api/src/domain/callback/index.types.d.ts

File: core/api/src/services/svix/index.ts [hunk 6]
```diff
@@ -31,15 +31,36 @@ function prefixObjectKeys(
...
+  sendMessage: (args: {
+    eventType: string
+    accountId: AccountId
+    walletId: WalletId
+    payload: Record<string, JSONValue>
+  }) => Promise<ApplicationError | true>
+  getPortalUrl: (accountId: AccountId) => Promise<ApplicationError | string>
```
Comment (dolcalmi) [inline core/api/src/services/svix/index.ts:41]: ApplicationError is from app layer, please create CallbackServiceError `export class CallbackServiceError extends CallbackError {}` here https://github.com/GaloyMoney/galoy/blob/main/core/api/src/domain/callback/errors.ts

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: core/api/src/services/svix/index.ts (hunk 5 above)
Comment (dolcalmi) [reply core/api/src/services/svix/index.ts:34]: also please rename to `ICallbackService`

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**